### PR TITLE
fix(build-tools): Fix dependency and task prioritization

### PR DESCRIPTION
**Pull request by curtisman**
Fix some of the missing dependencies
- `webpack` requires dependent packages `tsc` for typing
- `webpack` task in `PropertyDDS`'s `partial-checkout` and `property-inspector` require `build:copy-resources` from `property-inspector-table` just like it's `tsc` task.
- `test-dds-utils` package's `tsc` task depends on `stochastic-test-utils`'s `build:test` task

Formalize task prioritizing:
- In general, we want to make sure longer task, and tasks that will unblock the most tasks first, to make sure maximize concurrency and avoid a long tail of tasks at the end that doesn't parallelize well.
- Previous, we use the number of "parent" dependency task to estimate how much it will unblock and use that as the priority for scheduling.  However, this method can lead to double counting and doesn't account for the depth.
- Instead, the change, walk the "parent" dependency graph to collect all nodes, and sum up the weight on it.
- Each task can now specify the weight (for that that takes longer) so they will be schedule earlier to avoid long pole at the end.
- Benefit depends on the number core and task you are running.  An experiment with `-t compile --concurrent 5` yield around 3% speed up.

Other changes:
- Convert some of the verbose message to traces 
- Add `-q`/ `--quiet` switch to disable console output other than warning or errors, so I can focus on traces.
- Add `--concurrency` switch to control the number of tasks to run in parallel.
- Change  `LeafTask.collectDependentLeafTasks` that require a set to be passed into to `LeafTask.transitiveDependentLeafTask` to just return an array to make it more readable (there is no benefit in passing in a set to add to)

### CODE REVIEW

File: build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts [hunk 1]
```diff
@@ -79,6 +75,52 @@ export abstract class LeafTask extends Task {
...
+		if (this._parentLeafTasks === undefined) {
+			const parentLeafTasks = new Set<LeafTask>(this.directParentLeafTasks);
+			this._parentLeafTasks = parentLeafTasks;
+			this.directParentLeafTasks
+				.map((task) => task.parentLeafTasks)
+				.forEach((p) => p.forEach((t) => parentLeafTasks.add(t)));
+		}
+		return this._parentLeafTasks;
+	}
+	// End function used by initialWeight phase
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts:104]: Nit: Does this comment maybe indicate things could be better factored?

File: build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/miscTasks.ts [hunk 2]
```diff
@@ -14,12 +14,18 @@ import { LeafTask, LeafWithDoneFileTask, LeafWithFileStatDoneFileTask } from "./
 const { verbose } = defaultLogger;
 
 export class EchoTask extends LeafTask {
+	protected get taskWeight() {
+		return 0; // generally cheap relative to other tasks
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/miscTasks.ts:18]: Worth declaring some constants for common weights?

**Review by tylerbutler — APPROVED** [review]

File: build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts (hunk 1 above)
Comment (curtisman) [reply build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts:104]: No.  It's a left over comment.  I will remove.

File: build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/miscTasks.ts (hunk 2 above)
Comment (curtisman) [reply build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/miscTasks.ts:18]: `LeafTask.taskWeight` is the default (1)

File: build-tools/packages/build-tools/src/fluidBuild/fluidBuild.ts [hunk 3]
```diff
@@ -144,7 +144,9 @@ async function main() {
 				log(
 					`Execution time: ${totalElapsedTime.toFixed(
 						3,
-					)}s, Concurrency: ${concurrency.toFixed(3)}`,
+					)}s, Concurrency: ${concurrency.toFixed(
+						3,
+					)}, Queue Wait time: ${buildGraph.totalQueueWaitTime.toFixed(3)}s`,
```
Comment (alexvy86) [inline build-tools/packages/build-tools/src/fluidBuild/fluidBuild.ts:149]: Why, prettier, _why_!? 😭

**Pull request merged**